com.unity.render-pipelines.core/Editor/MaterialUpgrader.cs

```diff
@@ -392,15 +407,44 @@ public static void Upgrade(Material material, MaterialUpgrader upgrader, Upgrade
         /// <param name="flags">Material Upgrader flags.</param>
         public static void Upgrade(Material material, List<MaterialUpgrader> upgraders, UpgradeFlags flags)
         {
-            if (material == null)
+            string message = String.Empty;
+            if (Upgrade(material, upgraders, flags, ref message))
                 return;
 
+            if (!string.IsNullOrEmpty(message))
+            {
+                Debug.Log(message);
+            }
+        }
+
+        /// <summary>
+        /// Upgrade a material.
+        /// </summary>
+        /// <param name="material">Material to upgrade.</param>
+        /// <param name="upgraders">List of Material upgraders.</param>
+        /// <param name="flags">Material Upgrader flags.</param>
+        /// <param name="message">Reference to the message if the flags <see cref="UpgradeFlags.LogMessageWhenNoUpgraderFound"/> is used.</param>
```

Comment:
Could use better explanation. Error message to be outputted when no material upgraders are suitable for given material?

Reply:
agreed, Done